source/core/Ky.ts

```diff
@@ -118,9 +118,10 @@ export class Ky {
 	// eslint-disable-next-line complexity
 	constructor(input: Input, options: Options = {}) {
 		this._input = input;
+		const isCredentialsSupported = 'credentials' in Request.prototype;
 		this._options = {
 			// TODO: credentials can be removed when the spec change is implemented in all browsers. Context: https://www.chromestatus.com/feature/4539473312350208
-			credentials: (this._input as Request).credentials || 'same-origin',
+			credentials: isCredentialsSupported ? ((this._input as Request).credentials || 'same-origin') : 'same-origin',
```

Comment:
I don’t think this is gonna work, `_options.credentials` should not be set at all, according to the error.

Also the comment is very outdated and we can probably use:

```suggestion
			credentials: isCredentialsSupported ? (this._input as Request).credentials : undefined,
```

… as long as CloudFlare accepts `{credentials: undefined}` and doesn't actually require `{}`